core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -141,19 +142,20 @@ object ZkUtils {
   def getDeleteTopicPath(topic: String): String =
     DeleteTopicsPath + "/" + topic
 
-  // Parses without deduplicating keys so the data can be checked before allowing reassignment to proceed
   def parsePartitionReassignmentData(jsonData: String): Map[TopicAndPartition, Seq[Int]] = {
+    val parseResult = Json.parseStringAs[PartitionAssignment](jsonData)
+
+    val assignments = parseResult match {
+      case Left(throwable) => throw new ConfigException(s"Invalid reassignment config: $throwable")
```

Comment:
Seems like we'd want to include the original exception as the cause. Also, why is it a `ConfigException`?